AP_Scripting, AC_PosControl: add follow-target-send.lua applet

rmackay9 opened the pull request:
This adds a follow-target-send.lua applet which can send the [FOLLOW_TARGET](https://github.com/ArduPilot/mavlink/blob/master/message_definitions/v1.0/common.xml#L5532) mavlink message to other vehicles.  This will be used in the cooperative lift and/or copter-ship-landing work.

This has been lightly tested in SITL and below is a screenshot from MP's MAVLink inspector sent from a vehicle flying North at 10m/s in Guided mode
![follow-target-send-16Dec2024](https://github.com/user-attachments/assets/a9c94c35-bcf0-4c95-ad8c-93572509bf06)

I was hoping to test [Follow mode](https://ardupilot.org/copter/docs/follow-mode.html) with [multiple vehicles in SITL](https://ardupilot.org/dev/docs/using-sitl-for-ardupilot-testing.html#swarming-with-sitl) but it looks like mavlink messages are not passed between them.

BTW the mavlink spec for the [FOLLOW_TARGET](https://github.com/ArduPilot/mavlink/blob/master/message_definitions/v1.0/common.xml#L5532) message is not too clear on the frame of each field.  I've assumed everything is NED

IamPete1 commented on libraries/AP_Scripting/docs/docs.lua:
```diff
@@ -3785,6 +3785,18 @@ function poscontrol:set_posvelaccel_offset(pos_offset_NED, vel_offset_NED, accel
+---@return Vector3f_ud|nil
+function poscontrol:get_vel_target() end
+
+-- get position controller's acceleration velocity in m/s/s in NED frame
```
```suggestion
-- get position controller's target acceleration in m/s/s in NED frame
```

rmackay9 replied:
thanks, fixed!

IamPete1 commented on libraries/AP_Scripting/applets/follow-target-send.lua:
```diff
@@ -0,0 +1,119 @@
+
+    -- prepare FOLLOW_TARGET message
+    local follow_target_msg = {}
+    follow_target_msg.timestamp = millis():toint() * 1000
```
I think millis is correct, no need for the * 1000?

We do support 64bit numbers now. We could add a binding for millis64.

rmackay9 replied:
thanks @IamPete1 for the review.  I thought of this but then wondered if the millis64 wrapped more quickly than millis()?  Not that most vehicles are powered up long enough to wrap but..

rmackay9 replied:
oh my, you're right!  fixed!

rmackay9 commented:
I've fixed up issues found by @IamPete1 (thanks!).

I've also change the script slightly so that the capabilities are filled in dynamically and change with the flight mode.  E.g. in Stabilize the capability is "9" meaning position, attitude and rates while in Guided it is 15 meaning position, vel target, accel target, attitude and rates

Finally I've also rebased on master
